sqlmesh/core/scheduler.py

```diff
@@ -377,6 +380,26 @@ def earliest_start_date(snapshots: t.Iterable[Snapshot]) -> datetime:
     return yesterday()
 
 
+def latest_end_date(snapshots: t.Iterable[Snapshot]) -> datetime | date:
+    """Get the latest end date from a collection of snapshots.
+
+    Args:
+        snapshots: Snapshots to find latest end date.
+    Returns:
+        The latest end date or now if none is found."""
+    snapshots = list(snapshots)
+    if not any(snapshot.intervals for snapshot in snapshots):
+        return now()
+    if snapshots:
+        end_date, interval = max(
+            (snapshot.intervals[-1][1], snapshot.model.interval_unit()) for snapshot in snapshots
```

Comment:
if snapshot.intervals

Reply:
store the intervals_and_units, and check that there are none here and return now() as opposed to 391